src/README.md

```diff
@@ -0,0 +1,224 @@
+# Toit C++ style guide
+
+When contributing code to Toit, please follow the style of the
+existing C++ code as much as possible.  This makes code easier
+to read and update for other project members.
+
+Here we mention some of the style rules that we try to follow.
+In general it is more important to follow the
+style of the code already in Toit than to adhere to the style
+guide.
+
+## Language features
+
+For performance reasons, we compile with `-fno-exceptions` and so C++ exception
+handling is not available.
+
+We also compile with `-fno-rtti`, though this might change at some point.
+
+We do not allow use of the C++ standard library on the device, for reasons of
+code size.  In particular this means no use of `std::string`.
```

Comment:
Not just. `std::string` uses `malloc` which is a more important reason not to use it.